test/extended/two_node/tna_recovery.go

```diff
@@ -0,0 +1,168 @@
+	"time"
+
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+
+	v1 "github.com/openshift/api/config/v1"
+	exutil "github.com/openshift/origin/test/extended/util"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+var _ = g.Describe("[sig-etcd][apigroup:config.openshift.io][OCPFeatureGate:HighlyAvailableArbiter][Suite:openshift/two-node][Disruptive] Single node outage is handled seamlessly", func() {
+	defer g.GinkgoRecover()
+	oc := exutil.NewCLIWithoutNamespace("").AsAdmin()
+
+	g.BeforeEach(func() {
+		skipIfNotTopology(oc, v1.HighlyAvailableArbiterMode)
+	})
+
+	g.It("should maintain etcd quorum and workloads with one master node down", func() {
+		ctx := context.Background()
+
+		g.By("Identifying one master node to simulate failure")
+		masterNodes, err := oc.AdminKubeClient().CoreV1().Nodes().List(ctx, metav1.ListOptions{
+			LabelSelector: labelNodeRoleMaster,
+		})
+		o.Expect(err).To(o.BeNil())
+		o.Expect(len(masterNodes.Items)).To(o.Equal(2))
+		targetNode := masterNodes.Items[0].Name
+
+		g.By(fmt.Sprintf("Gracefully rebooting %s to simulate failure", targetNode))
+		shutdownOrRebootNode(oc, targetNode, "openshift-etcd", "shutdown", "-r", "+1")
+
+		g.By("Waiting for the node to become NotReady")
+		waitForNodeCondition(oc, targetNode, corev1.NodeReady, corev1.ConditionFalse, "NotReady", 10*time.Minute)
+
+		g.By("Validating etcd quorum is met while the node is still NotReady")
+		err = wait.PollUntilContextTimeout(ctx, 5*time.Second, 2*time.Minute, true, func(ctx context.Context) (bool, error) {
+			node, err := oc.AdminKubeClient().CoreV1().Nodes().Get(ctx, targetNode, metav1.GetOptions{})
+			if err != nil {
+				return false, nil
+			}
+			stillNotReady := false
+			for _, cond := range node.Status.Conditions {
+				if cond.Type == corev1.NodeReady && cond.Status != corev1.ConditionTrue {
+					stillNotReady = true
+					break
+				}
+			}
+			if !stillNotReady {
+				return false, nil
+			}
+
+			operator, err := oc.AdminConfigClient().ConfigV1().ClusterOperators().Get(ctx, "etcd", metav1.GetOptions{})
+			if err != nil {
+				return false, nil
+			}
+			return isClusterOperatorAvailable(operator), nil
+		})
+		o.Expect(err).To(o.BeNil(), "Expected etcd operator to remain healthy while one master node is NotReady")
+	})
+	g.AfterEach(func() {
+		ctx := context.Background()
+		g.By("Ensuring all cluster nodes are back to Ready state")
+
+		nodeList, err := oc.AdminKubeClient().CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+		o.Expect(err).To(o.BeNil(), "Failed to list cluster nodes")
+
+		for _, node := range nodeList.Items {
+			waitForNodeCondition(oc, node.Name, corev1.NodeReady, corev1.ConditionTrue, "Ready", 15*time.Minute)
+		}
+	})
+})
+
+var _ = g.Describe("[sig-etcd][apigroup:config.openshift.io][OCPFeatureGate:HighlyAvailableArbiter][Suite:openshift/two-node][Disruptive] Recovery after arbiter down and master node restart", func() {
```

Comment:
Same idea here, maybe change this to `Recovery when arbiter node is down and master nodes restart`